x/mongo/driver/wiremessage/tranquil_msgheader.go

```diff
@@ -0,0 +1,33 @@
+package wiremessage
+
+import "errors"
+
+// DefaultSize defines the expected message length received by the server.
+// This constant should be use when allocating a byte slice for a new message.
+// Note that in some places of the code uses a value of 256.
+const DefaultSize = 512
+
+// MsgHeader represents the header of a wire message. All messages
+// generated by clients and the server use this header.
+type MsgHeader struct {
+	Length     int32
+	RequestID  int32
+	ResponseTo int32
+	Opcode     OpCode
+}
+
+// ReadMsgHeader reads a wire message header from src.
+func ReadMsgHeader(src []byte) (*MsgHeader, error) {
+	var ok bool
+	hdr := &MsgHeader{}
+	hdr.Length, hdr.RequestID, hdr.ResponseTo, hdr.Opcode, _, ok = ReadHeader(src)
+	if !ok {
+		return nil, errors.New("Incomplete header")
+	}
+	return hdr, nil
+}
+
+// WriteMsgHeader reads a wire message header from src.
```

Comment:
doesn't `WriteMsgHeader` write a message header to `dst`? I think the comment should include the important fact that the header needs to be fully populated before `WriteMsgHeader` is called.

Reply:
Very good point. Will update the comment. 

I suspect the writing of the header will evolve and I get a better feel - and write a "wrapper" for writing out the "body" - i.e. json.Marshal()-ed operation.